File: frontend/src/service/model/audit.ts
import { baseUrl } from "service/common/api"
import { FilterRequest } from "service/common/filter";
import { paginate, Paginated } from "service/common/paginate";
import { SortRequest } from "service/common/sort";
import { catchError } from "service/error/catch";
import { APIError } from "service/error/model";
import { Audit } from "service/model/model";

export const getAudits = async (
    token: string,
    page?: number,
    size?: number,
    filter?: FilterRequest,
    sort?: SortRequest,
): Promise<Paginated<Audit> | APIError> => {
    return await catchError<Paginated<Audit>>(async () => {
        return await paginate({
            url: baseUrl() + '/audits',
            token: token,
            page: page,
            size: size,
            filter: filter,
            sort: sort,
        });    
    });
}

File: backend/internal/http/handler/audit.go
package handler

import (
	"net/http"

	"github.com/eko/authz/backend/internal/entity/manager"
	"github.com/eko/authz/backend/internal/entity/repository"
	"github.com/eko/authz/backend/internal/http/handler/model"
	"github.com/gofiber/fiber/v2"
)

// Retrieve audits for last days
//
//	@security	Authentication
//	@Summary	Retrieve audits for last days
//	@Tags		Check
//	@Produce	json
//	@Param		page	query		int		false	"page number"			example(1)
//	@Param		size	query		int		false	"page size"				minimum(1)	maximum(1000)	default(100)
//	@Param		filter	query		string	false	"filter on a field"		example(kind:contains:something)
//	@Param		sort	query		string	false	"sort field and order"	example(kind:desc)
//	@Success	200		{object}	[]model.Audit
//	@Failure	400		{object}	model.ErrorResponse
//	@Failure	500		{object}	model.ErrorResponse
//	@Router		/v1/audits [Get]
func AuditGet(
	auditManager manager.Audit,
) fiber.Handler {
	return func(c *fiber.Ctx) error {
		page, size, err := paginate(c)
		if err != nil {
			return returnError(c, http.StatusInternalServerError, err)
		}

		audits, total, err := auditManager.GetRepository().Find(
			repository.WithPage(page),
			repository.WithSize(size),
			repository.WithFilter(httpFilterToORM(c)),
			repository.WithSort(httpSortToORM(c)),
		)
		if err != nil {
			return returnError(c, http.StatusInternalServerError, err)
		}

		return c.JSON(model.NewPaginated(audits, total, page, size))
	}
}
